braindecode/models/usleep.py

```diff
@@ -0,0 +1,281 @@
+# Authors: Theo Gnassounou <theo.gnassounou@inria.fr>
+#          Omar Chehab <l-emir-omar.chehab@inria.fr>
+#
+# License: BSD (3-clause)
+
+import numpy as np
+import torch
+from torch import nn
+
+
+def crop_tensors_to_match(x1, x2, axis=-1):
+    '''Crops two tensors to their lowest-common-dimension along an axis.'''
+    dim_cropped = min(x1.shape[axis], x2.shape[axis])
+
+    x1_cropped = torch.index_select(
+        x1, dim=axis,
+        index=torch.arange(dim_cropped).to(device=x1.device)
+    )
+    x2_cropped = torch.index_select(
+        x2, dim=axis,
+        index=torch.arange(dim_cropped).to(device=x1.device)
+    )
+    return x1_cropped, x2_cropped
+
+
+class EncoderBlock(nn.Module):
```

Comment:
```suggestion
class _EncoderBlock(nn.Module):
```

make it private. Just keep USleep public